src/Agent.Plugins/PipelineCache/TarUtils.cs

```diff
@@ -195,24 +192,57 @@ public static class TarUtils
             }
         }
 
-        private static void SetProcessStartInfo(Process process, string processFileName, string processArguments)
+        private static void SetProcessStartInfo(ProcessStartInfo processStartInfo, string processFileName, string processArguments, string processWorkingDirectory)
+        {
+            processStartInfo.FileName = processFileName;
+            processStartInfo.Arguments = processArguments;
+            processStartInfo.UseShellExecute = false;
+            processStartInfo.RedirectStandardInput = true;
+            processStartInfo.RedirectStandardOutput = true;
+            processStartInfo.RedirectStandardError = true;
+            processStartInfo.WorkingDirectory = processWorkingDirectory;          
+        }
+
+        private static ProcessStartInfo GetCreateTarProcessInfo(string archiveFileName, string inputPath)
         {
-            process.StartInfo.FileName = processFileName;
-            process.StartInfo.Arguments = processArguments;
-            process.StartInfo.UseShellExecute = false;
-            process.StartInfo.RedirectStandardInput = true;
-            process.StartInfo.RedirectStandardOutput = true;
-            process.StartInfo.RedirectStandardError = true;
+            var processFileName = "tar";
+            inputPath = (inputPath.EndsWith(Path.DirectorySeparatorChar)) ? inputPath.TrimEnd(Path.DirectorySeparatorChar) : inputPath;
+            var processArguments = $"-cf \"{archiveFileName}\" -C \"{inputPath}\" ."; // If given the absolute path for the '-cf' option, the GNU tar fails. The workaround is to start the tarring process in the temp directory, and simply speficy 'archive.tar' for that option.
+            ProcessStartInfo processStartInfo = new ProcessStartInfo();
+            SetProcessStartInfo(processStartInfo, processFileName, processArguments, processWorkingDirectory: Path.GetTempPath());
```

Comment:
>Path.GetTempPath() [](start = 110, length = 18)

explain through a comment why processWorkingDirectory for TAR is the TEMP path directory. I remember you explained it to me but for someone else, it may need some explanation.

Reply:
Added comment.